From: Richard Acayan <mailingradian@gmail.com>
Date: Thu, 7 Aug 2025 17:14:02 -0400
Subject: [PATCH] dt-bindings: pinctrl: qcom: Add SDM670 LPI pinctrl

Signed-off-by: Richard Acayan <mailingradian@gmail.com>
---
 ...trl.yaml => qcom,sdm670-lpass-lpi-pinctrl.yaml} | 60 ++++++----------------
 1 file changed, 17 insertions(+), 43 deletions(-)
 copy Documentation/devicetree/bindings/pinctrl/{qcom,sdm660-lpass-lpi-pinctrl.yaml => qcom,sdm670-lpass-lpi-pinctrl.yaml} (51%)

diff --git a/Documentation/devicetree/bindings/pinctrl/qcom,sdm660-lpass-lpi-pinctrl.yaml b/Documentation/devicetree/bindings/pinctrl/qcom,sdm670-lpass-lpi-pinctrl.yaml
similarity index 51%
copy from Documentation/devicetree/bindings/pinctrl/qcom,sdm660-lpass-lpi-pinctrl.yaml
copy to Documentation/devicetree/bindings/pinctrl/qcom,sdm670-lpass-lpi-pinctrl.yaml
index 409e5a4..96d3f40 100644
--- a/Documentation/devicetree/bindings/pinctrl/qcom,sdm660-lpass-lpi-pinctrl.yaml
+++ b/Documentation/devicetree/bindings/pinctrl/qcom,sdm670-lpass-lpi-pinctrl.yaml
@@ -1,21 +1,23 @@
 # SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
 %YAML 1.2
 ---
-$id: http://devicetree.org/schemas/pinctrl/qcom,sdm660-lpass-lpi-pinctrl.yaml#
+$id: http://devicetree.org/schemas/pinctrl/qcom,sdm670-lpass-lpi-pinctrl.yaml#
 $schema: http://devicetree.org/meta-schemas/core.yaml#
 
-title: Qualcomm SDM660 SoC LPASS LPI TLMM
+title: Qualcomm SDM670 SoC LPASS LPI TLMM
 
 maintainers:
-  - Nickolay Goppen <setotau@mainlining.org>
+  - Richard Acayan <mailingradian@gmail.com>
+    # Maybe?
+  - Srinivas Kandagatla <srinivas.kandagatla@linaro.org>
 
 description:
   Top Level Mode Multiplexer pin controller in the Low Power Audio SubSystem
-  (LPASS) Low Power Island (LPI) of Qualcomm SDM660 SoC.
+  (LPASS) Low Power Island (LPI) of Qualcomm SDM670 SoC.
 
 properties:
   compatible:
-    const: qcom,sdm660-lpass-lpi-pinctrl
+    const: qcom,sdm670-lpass-lpi-pinctrl
 
   reg:
     items:
@@ -24,14 +26,14 @@ properties:
 patternProperties:
   "-state$":
     oneOf:
-      - $ref: "#/$defs/qcom-sdm660-lpass-state"
+      - $ref: "#/$defs/qcom-sdm670-lpass-state"
       - patternProperties:
           "-pins$":
-            $ref: "#/$defs/qcom-sdm660-lpass-state"
+            $ref: "#/$defs/qcom-sdm670-lpass-state"
         additionalProperties: false
 
 $defs:
-  qcom-sdm660-lpass-state:
+  qcom-sdm670-lpass-state:
     type: object
     description:
       Pinctrl node's client devices use subnodes for desired pin configuration.
@@ -45,11 +47,12 @@ $defs:
           List of gpio pins affected by the properties specified in this
           subnode.
         items:
-          pattern: "^gpio([0-9]|[1-2][0-9]|3[0-1])$"
+          pattern: "^gpio([0-9]|1[0-9]|2[0-9]|3[0-1])$"
 
       function:
         enum: [ gpio, comp_rx, dmic1_clk, dmic1_data, dmic2_clk, dmic2_data,
-                mclk0, pdm_tx, pdm_clk, pdm_rx, pdm_sync ]
+                lpi_cdc_rst, mclk0, pdm_tx, pdm_clk, pdm_rx, pdm_sync, sec_tdm,
+                sec_tdm_din, sec_tdm_dout ]
         description:
           Specify the alternative function to be configured for the specified
           pins.
@@ -65,45 +68,16 @@ unevaluatedProperties: false
 
 examples:
   - |
-    lpi_tlmm: pinctrl@15070000 {
-        compatible = "qcom,sdm660-lpass-lpi-pinctrl";
-        reg = <0x15070000 0x20000>;
+    lpi_tlmm: pinctrl@62b40000 {
+        compatible = "qcom,sdm670-lpass-lpi-pinctrl";
+        reg = <0 0x62b40000 0 0x20000>;
         gpio-controller;
         #gpio-cells = <2>;
         gpio-ranges = <&lpi_tlmm 0 0 32>;
 
-        cdc_pdm_default: cdc-pdm-default-state {
-            clk-pins {
-                pins = "gpio18";
-                function = "pdm_clk";
-                drive-strength = <8>;
-                output-high;
-            };
-
-            sync-pins{
-                pins = "gpio19";
-                function = "pdm_sync";
-                drive-strength = <4>;
-                output-high;
-            };
-
-            tx-pins {
-                pins = "gpio20";
-                function = "pdm_tx";
-                drive-strength = <8>;
-            };
-
-            rx-pins {
-                pins = "gpio21", "gpio23", "gpio25";
-                function = "pdm_rx";
-                drive-strength = <4>;
-                output-high;
-            };
-        };
-
         cdc_comp_default: cdc-comp-default-state {
             pins = "gpio22", "gpio24";
             function = "comp_rx";
-            drive-strength = <8>;
+            drive-strength = <4>;
         };
     };
